Category List API

EnTeQuAk opened the pull request:
Fixes https://github.com/mozilla/addons-server/issues/3676, supports https://github.com/mozilla/addons-frontend/pull/1237/

diox commented:
It's probably worth adding caching headers using `finalize_response()` in the view ? Since categories don't really change...

diox commented on src/olympia/constants/categories.py:
```diff
@@ -9,21 +9,25 @@
 
 
 class StaticCategory(object):
```
Somewhere in that class should the reason behind immutability described.

diox commented on src/olympia/constants/categories.py:
```diff
@@ -9,21 +9,25 @@
-        self.weight = weight
+        # Avoid triggering our own __setattr__ implementation
+        # to keep immutability intact but set initial values.
+        object.__setattr__(self, 'id', id)
```
Well, it's ugly but it does the job...

EnTeQuAk commented:
Updated docs, added caching. I'm going to merge this once travis is green, if there is anything that needs fixing / improving let's do that in a separate PR. Don't want to block @tofumatt any longer.
